core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -632,6 +634,7 @@ class KafkaServer(
           case t: Throwable => throw new RuntimeException("Received a fatal error while " +
             "waiting for the SocketServer Acceptors to be started.", t)
         }
+        info("End processing authorizer futures")
```

Comment:
`enableRequestProcessingFuture` is not related to `authorizer`. Also, I saw when in KRaft mode, we'll use `FutureUtils.waitWithLogging` [here](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/BrokerServer.scala#L530). Do you think we can use it? Why and why not?

Reply:
Thanks for your comment @showuon :smiley: , I have some questions here.

> enableRequestProcessingFuture is not related to authorizer

Not sure if I misunderstanding anything, but `enableRequestProcessingFuture` wrap the `authorizerFutures`, and KafkaServer will until enableRequestProcessingFuture finished. If we setup the incorrect ACL class, like you mentioned before, under zookeeper mode and set `authorizer.class.name=org.apache.kafka.metadata.authorizer.StandardAuthorizer`, the KafkaServer stuck [here](https://github.com/apache/kafka/blob/6f8d4fe26b7ceaa369c5e974a8cf54cc2714f9a7/core/src/main/scala/kafka/server/KafkaServer.scala#L630)  since KRaft is not enabled under zookeeper mode. 

Or did you mean the log in 637 is in the wrong place ? Looks like we should place it after L629 ?

> Also, I saw when in KRaft mode, we'll use FutureUtils.waitWithLogging [here](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/BrokerServer.scala#L530). Do you think we can use it? Why and why not?

That inspired me, in KRaft mode we have a configuration called `server.max.startup.time.ms`, maybe we can introduce the same config to Zookeeper mode to avoid kafka server hanging forever if something goes wrong. Do you think this is available ? If so, I'm willing to solve that !

Reply:
> Or did you mean the log in 637 is in the wrong place ? Looks like we should place it after L629 ?

Yes, this is what I meant. Speaking of `enableRequestProcessingFuture`, I think we should also add logs before and after `enableRequestProcessingFuture.join()` to avoid server hanging. WDYT?

> That inspired me, in KRaft mode we have a configuration called server.max.startup.time.ms, maybe we can introduce the same config to Zookeeper mode to avoid kafka server hanging forever if something goes wrong. Do you think this is available ? If so, I'm willing to solve that !

Thanks for looking into it. If the config only works in KRaft mode, then we cannot use that in ZK mode.
To introduce a new config needs to write a proposal (KIP) and open for discussion and vote. Given we're going to remove ZK from Kafka soon, I don't think it's worth adding it.

Reply:
> Yes, this is what I meant. Speaking of enableRequestProcessingFuture, I think we should also add logs before and after enableRequestProcessingFuture.join() to avoid server hanging. WDYT?

Will do, many thanks !

> Thanks for looking into it. If the config only works in KRaft mode, then we cannot use that in ZK mode.
To introduce a new config needs to write a proposal (KIP) and open for discussion and vote. Given we're going to remove ZK from Kafka soon, I don't think it's worth adding it.

Indeed... anyway, thanks for the comment !